spec/features/consumer/shopping/variant_overrides_spec.rb

```diff
@@ -146,6 +149,15 @@
       product1_variant1_override.reload.count_on_hand.should be_nil
     end
 
+    it "does not subtract stock from variants where the override has on_demand: true" do
```

Comment:
I really like how you wrote the test first and have it in a separate commit. :+1:

I think that it's even better to declare a spec as pending when you know that it fails because of a bug. So you can run the specs and they "pass". The commit that fixed the bug then changes the spec from `pending` to `it` and you clearly see in the commit history which change fixed what spec (and you can rely on CI and `git bisect` when needed). But maybe that's just my preference?